workspaces/adventure-pack/goodies/kotlin/src/traverse_level_order/Main.kt

```diff
@@ -0,0 +1,26 @@
+package traverse_level_order
+
+import common.TreeNode
+
+public fun TreeNode?.traverseLevelOrder(): Sequence<TreeNode> {
+    if (this == null) return sequenceOf()
+
+    val q = ArrayDeque<TreeNode>()
+    q.add(this)
+
+    return sequence {
+        while (q.isNotEmpty()) {
+            val current = q.removeFirst()
+
+            yield(current)
+
+            if (current.left != null) {
+                q.add(current.left!!)
```

Comment:
I am not familiar with Kotlin either, and I was curious whether the non-null assertion is necessary, given that within the `if` condition this is bound to be non-null. The same for line 22.